mypy/server/astdiff.py

```diff
@@ -280,6 +283,7 @@ def snapshot_definition(node: SymbolNode | None, common: SymbolSnapshot) -> Symb
             tuple(snapshot_type(tdef) for tdef in node.defn.type_vars),
             [snapshot_type(base) for base in node.bases],
             [snapshot_type(p) for p in node._promote],
+            node.dataclass_transform_spec or find_dataclass_transform_spec(node),
```

Comment:
this seems fairly expensive; an alternative may be to attach the relevant spec to each class in the heirarchy during semantic analysis rather than only attaching the spec to the transformer class

Reply:
Class hierarchies are almost always quite shallow, so this shouldn't be expensive. Similar to above, the snapshots must support equality. Using the serialized form should do the trick.